forge-gui/res/cardsfolder/upcoming/kellan_the_kid.txt

```diff
@@ -0,0 +1,13 @@
+Name:Kellan, the Kid
+ManaCost:G W U
+Types:Legendary Creature Human Faerie Rogue
+PT:3/3
+K:Flying
+K:Lifelink
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ DBChoice | TriggerDescription$ Whenever you cast a spell from anywhere other than your hand, you may cast a permanent spell with equal or lesser mana value from your hand without paying its mana cost. If you don't, you may put a land card from your hand onto the battlefield.
+SVar:DBChoice:DB$ GenericChoice | Defined$ You | Choices$ CastSpell,PlayLand
```

Comment:
My gut says skip GenericChoice and go straight to Optional DB$ Play + RememberPlayed
Then SubAbility `PlayLand:DB$ ChangeZone` with Condition
"Baral and Kari Zev" is a good example 

That said, this GenericChoice way might be better for the AI to consider both options equally? @Agetian ?

Might be good to have a "Neither" choice too?

Reply:
I'm gonna say, this worked adequately enough I didn't have to iterate much to resolve issues. 
Also the structure of that part of the script was lifted from Spark of Creativity which is closer since the choices for both Spark and Kellan are optional while for Baral and Kari Zev creating the Ragavan token isn't.

Reply:
Often there is more than one way to do these sorts of branching effects. I'm just wondering if AI might be "smarter" in one case or the other, hence calling in Agetian 😁

Reply:
AI doesn't really understand GenericChoice this way
Also if you want to to cast a spell it feels weird you'd have to confirm twice here?

Reply:
Northmoc had it right in recommending Baral and Kari Zev as a template. I ran across a corner case where there's no relevant spell to cast for free in my hand but there's a land to play: misplaying into selecting the option to cast the spell will just fizzle the ability with no further effect. So I lifted the script from Baral and Kari Zev so it defaults into a land play in that case, still with all steps being optional.